src/common.c

```diff
@@ -264,6 +264,17 @@ const char *ofi_straddr(char *buf, size_t *len,
 	case FI_ADDR_MLX:
 		size = snprintf(buf, *len, "fi_addr_mlx://%p", addr);
 		break;
+	case FI_ADDR_IB_UD:
+		size = snprintf(buf, *len, "fi_addr_ib_ud://"
+				"%" PRIx16 /* LID */ ":%" PRIx8 /* SL */
+				":%" PRIx64 /* GID::GUID */ ":%" PRIx64 /* GID::EUI */
+				":%" PRIx32 /* QPN */ ":%" PRIx32 /* PSN */ ":%" PRIx16 /* P_Key */
+				":%" PRIx16 /* NS service */,
+				*(uint16_t *)addr + 12, *(uint8_t *)addr + 30,
+				*(uint64_t *)addr, *(uint64_t *)addr + 1,
+				*(uint32_t *)addr + 4, *(uint32_t *)addr + 5, *(uint16_t *)addr + 13,
+				*(uint16_t *)addr + 14);
```

Comment:
Thanks for reviewing this. I'll update
Did you mean the following (`%` -> `:`):
`fi_addr_ib_ud://[gid]:qpn/pkey/lid/sl` ?
I've not found "%" for usage in URI.